src/BTVNanoCommissioning/utils/correction.py

```diff
@@ -423,36 +540,54 @@ def JME_shifts(
                 ]
                 if len(jecname) > 1:
                     raise ("Multiple uncertainties match to this era")
+                elif len(jecname) == 0:
+                    raise ("no uncertainty match to this era")
                 else:
                     jecname = jecname[0] + "_DATA"
             else:
-                jecname = correct_map["JME_cfg"]["MC"] + "_MC"
-            corr = correct_map["JME"].compound[f"{jecname}_L1L2L3Res_AK4PFPuppi"]
+                jecname = correct_map["JME_cfg"]["MC"].split(" ")[0] + "_MC"
+                jrname = correct_map["JME_cfg"]["MC"].split(" ")[1] + "_MC"
+
+            # store the original jet info
             nocorrjet = events.Jet
             nocorrjet["pt_raw"] = (1 - nocorrjet["rawFactor"]) * nocorrjet["pt"]
             nocorrjet["mass_raw"] = (1 - nocorrjet["rawFactor"]) * nocorrjet["mass"]
             nocorrjet["rho"] = ak.broadcast_arrays(
                 events.fixedGridRhoFastjetAll, nocorrjet.pt
             )[0]
-            j, nj = ak.flatten(nocorrjet), ak.num(nocorrjet)
-            values = [
-                np.array(
-                    j[
-                        inputs.name.replace("Jet", "")
-                        .replace("Pt", "pt")
-                        .replace("Phi", "phi")
-                        .replace("Eta", "eta")
-                        .replace("Mass", "mass")
-                        .replace("Rho", "rho")
-                        .replace("A", "area")
-                    ]
-                )
-                for inputs in corr.inputs
-            ]
-            flatCorrFactor = corr.evaluate(*values)
-            corrFactor = ak.unflatten(flatCorrFactor, nj)
+            nocorrjet["EventID"] = ak.broadcast_arrays(events.event, nocorrjet.pt)[0]
+            nocorrjet["Genpt"] = events.GenJet[nocorrjet.genJetIdx].pt
```

Comment:
This should only run for MC (I think):

```suggestion
            if not isRealData: nocorrjet["Genpt"] = events.GenJet[nocorrjet.genJetIdx].pt
```

Reply:
Hi @philippgadow thanks for the suggestion, you are right, I need to fix this :p I was only test on MC and this does not show error messages